File: Service/Admin/RMA.Service.Admin.MasterDataManager.Contracts/Entities/UserReminderSearchRequest.cs

using RMA.Common.Entities.DatabaseQuery;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.Admin.MasterDataManager.Contracts.Entities
{
    public class UserReminderSearchRequest
    {
        public int UserId { get; set; }
        public List<UserReminderTypeEnum> UserReminderTypes { get; set; }
        public UserReminderItemTypeEnum? UserReminderItemType { get; set; }
        public int? ItemId { get; set; }
        public bool GetAlerts { get; set; }

        public PagedRequest PagedRequest { get; set; }

        public DateTime? StartDateFilter { get; set; }
        public DateTime? EndDateFilter { get; set; }
        public List<int?> UsersFilter { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/shared-models-lib/src/lib/user-reminders-and-messages/user-reminder-search-request.ts
import { PagedRequest } from "../pagination/PagedRequest";
import { UserReminderItemTypeEnum } from "./user-reminder-item-type-enum";
import { UserReminderTypeEnum } from "./user-reminder-type-enum";

export class UserReminderSearchRequest {
    userId: number;
    userReminderTypes: UserReminderTypeEnum[];
    userReminderItemType: UserReminderItemTypeEnum;
    itemId: number;
    getAlerts: boolean;

    pagedRequest: PagedRequest;

    startDateFilter: Date;
    endDateFilter: Date;

    usersFilter: number[];
}
